packages/fx-core/src/component/driver/botFramework/createOrUpdateBot.ts

```diff
@@ -75,13 +81,29 @@ export class CreateOrUpdateBotFrameworkBotDriver implements StepDriver {
         getLocalizedString("driver.botFramework.progressBar.createOrUpdateBot")
       );
 
+      let callingEndpoint: string | undefined = undefined;
+      let configuredChannels: BotChannelType[] | undefined = undefined;
+      if (args.channels) {
```

Comment:
What if `channels` is undefined?
And what's the case to set `enable` to false for certain channel?

Reply:
If `channels` is undefined, will not update the configured channels, see https://github.com/OfficeDev/TeamsFx/blob/kuojianlu/botFramework-channel/packages/fx-core/src/component/resource/botService/botRegistration/utils.ts#L18.
If setting `enable` to false, will disable the channel.

Reply:
How about removing `enable`? To disable the channel, just to remove the whole object?

Reply:
Good idea, `enable` seems redundant. Let me update it.